### PR TITLE
Support length-weighted normalization in loss calculation

### PR DESCRIPTION
**Pull request by kan-bayashi**
This PR supports length-weighted normalization in loss calculation.
From #1360, it seems that the length weighted normalized loss is effective to get clear diagonal attention.

### Now
loss (B, T, D) -> average over all of the bins -> loss scalar

### length-weighted normalization
loss (B, T, D) -> average along with T -> loss (B, D) -> average over remaining bins -> loss scalar

### CODE REVIEW

**Comment by kan-bayashi** [comment]
I added results. 
From this result, I also included ASR eval results.
This is nice to compare, I think.
It is worthwhile to try the dataset where there are various length samples.
In the case of the LJspeech, the results are almost the same between v3 and v4.

**Comment by kan-bayashi** [comment]
@r9y9 Could you check it?
I think not so important option but it is worthwhile to try as we can see in #1360.

File: espnet/nets/pytorch_backend/e2e_tts_tacotron2.py [hunk 1]
```diff
@@ -160,17 +160,26 @@ def _make_masks(ilens, olens):
...
     """Loss function module for Tacotron2."""
 
-    def __init__(self, use_masking=True, bce_pos_weight=20.0):
+    def __init__(self, use_masking=True, use_weighted_masking=False, bce_pos_weight=20.0):
         """Initialize Tactoron2 loss module.
 
         Args:
-            use_masking (bool): Whether to mask padded part in loss calculation.
+            use_masking (bool): Whether to apply masking for padded part in loss calculation.
+            use_weighted_masking (bool): Whether to applyg weighted for maskng padded part in loss calculation.
```
Comment (r9y9) [inline espnet/nets/pytorch_backend/e2e_tts_tacotron2.py]: Same as above

File: espnet/nets/pytorch_backend/e2e_tts_transformer.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 from espnet.nets.pytorch_backend.e2e_asr_transformer import subsequent_mask
 from espnet.nets.pytorch_backend.e2e_tts_tacotron2 import GuidedAttentionLoss
+from espnet.nets.pytorch_backend.e2e_tts_tacotron2 import Tacotron2Loss as TransformerLoss
```
Comment (r9y9) [inline espnet/nets/pytorch_backend/e2e_tts_transformer.py]: Oh :)

We could consider naming it a more clear name that denotes what it actually does. Not an issue of the PR though.

File: espnet/nets/pytorch_backend/e2e_tts_transformer.py [hunk 3]
```diff
@@ -325,7 +277,8 @@ def __init__(self, idim, odim, args=None):
                 - eprenet_dropout_rate (float): Dropout rate in encoder prenet.
                 - dprenet_dropout_rate (float): Dropout rate in decoder prenet.
                 - postnet_dropout_rate (float): Dropout rate in postnet.
-                - use_masking (bool): Whether to use masking in calculation of loss.
+                - use_masking (bool): Whether to apply masking for padded part in loss calculation.
+                - use_weighted_masking (bool): Whether to apply weighted masking for padded part in loss calculation.
```
Comment (r9y9) [inline espnet/nets/pytorch_backend/e2e_tts_transformer.py]: Same as the above; a bit confusing. Weighted mask (`w`) is applied for the non-padded part as well (0 is for the padded part) in my understanding.

Comment (kan-bayashi) [reply espnet/nets/pytorch_backend/e2e_tts_transformer.py]: Right. I removed `for padded part`.

**Comment by kan-bayashi** [comment]
Added all of the ASR results.

**Pull request merged**